lib/WIFI/WIFI.cpp

```diff
@@ -0,0 +1,299 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+/**************************************************************************************************
+* File: WLAN.cpp
+* @brief: Implementation of WLAN.h.
+* @author: Gabryel Reyes <gabryelrdiaz@gmail.com>
+**************************************************************************************************/
+/* INCLUDES **************************************************************************************/
+#include "WIFI.h"
+#include <EEPROM.h>
+
+/* CONSTANTS *************************************************************************************/
+
+static const unsigned long WIFI_TIMEOUT_MS = 5000; /**< Timeout for WiFi connection. */
+static const uint8_t CREDENTIALS_MAX_LENGTH = 50;  /**< Maximum length for saved Credentials. */
+static const uint8_t NVM_SSID_ADDRESS = 0;         /**< Address of saved SSID in EEPROM. */
+static const uint8_t NVM_PASSWORD_ADDRESS = 50;    /**< Address of saved Password in EEPROM. */
+
+/* MACROS ****************************************************************************************/
+
+/* TYPES *****************************************************************************************/
+
+/* PROTOTYPES ************************************************************************************/
+
+/* VARIABLES *************************************************************************************/
+
+static bool staAvailable = false; /**< True if system is connected in STA mode to a Network. */
+
+/* PUBLIC METHODES *******************************************************************************/
+
+/**************************************************************************************************/
+
+/**
+ *  Default Constructor.
+ */
+WIFI::WIFI() : AP_SSID("RacingLapTimer"), AP_PASSWORD("let me in"), STA_SSID(""), STA_PASSWORD("")
+{
+}
+
+/**************************************************************************************************/
+
+/**
+ *  Default Destructor.
+ */
+WIFI::~WIFI()
+{
+}
+
+/**************************************************************************************************/
+
+/**
+ *  Initialize WIFI Module.
+ * 
+ *  @return success.
+ */
+bool WIFI::begin()
+{
+    bool success = false;
+
+    EEPROM.begin(512);
```

Comment:
512 = Magic number, used at several places.

Think about moving EEPROM stuff to a separate class.